cli/src/semdep/parse_lockfile.py

```diff
@@ -637,27 +594,41 @@ def depth(package_index: int) -> int:
     "cargo.lock": parse_cargo,  # Rust
     "maven_dep_tree.txt": parse_pom_tree,  # Java
     "gradle.lockfile": parse_gradle,  # Java
-    "poetry.lock": parse_poetry,  # Python
     "requirements.txt": parse_requirements,
 }
 
+NEW_LOCKFILE_PARSERS = {
+    "poetry.lock": parse_poetry,  # Python
+}
+
 
 @lru_cache(maxsize=1000)
-def parse_lockfile_str(
-    lockfile_text: str, filepath_for_reference: Path, manifest_text: Optional[str]
+def parse_lockfile_path(
+    lockfile_path: Path, manifest_path: Optional[Path]
 ) -> List[FoundDependency]:
     # coupling with the github action, which decides to send files with these names back to us
-    filepath = filepath_for_reference.name.lower()
-    if filepath in LOCKFILE_PARSERS:
+    lockfile_name = lockfile_path.name.lower()
+    if lockfile_name in NEW_LOCKFILE_PARSERS:
+        return NEW_LOCKFILE_PARSERS[lockfile_name](lockfile_path, manifest_path)
```

Comment:
```suggestion
        parse_lockfile = NEW_LOCKFILE_PARSERS[lockfile_name]
        return parse_lockfile(lockfile_path, manifest_path)
```

I find this is more readable than `foo[bar](baz)`

Reply:
Sounds good to me, I'm neutral